airflow/providers/apache/beam/hooks/beam.py

```diff
@@ -94,6 +94,7 @@ def __init__(
         self.log.info("Running command: %s", " ".join(shlex.quote(c) for c in cmd))
         self.process_line_callback = process_line_callback
         self.job_id: Optional[str] = None
+        # pylint: disable=consider-using-with
```

Comment:
This is a bit sad because this class should be a function. We don't need to keep the state here.

Reply:
This class is only used once in Airflow code and the method of that class is called immediately after that, so converting it to a regular function shouldn't be difficult.
https://github.com/apache/airflow/blob/d598630cd7a2e80fd84c499cfcba37ad2b6888f3/airflow/providers/apache/beam/hooks/beam.py#L181-L185

Reply:
Possibly we can rewrite it, but this can be done separately. I wanted to keep minimal/safe set of changes to make it works without risking backwards compatibility/unforeseen side effects.

Reply:
I created issue for it https://github.com/apache/airflow/issues/15536